libs/libc/sched/task_setcanceltype.c

```diff
@@ -46,14 +54,70 @@
 
 int task_setcanceltype(int type, FAR int *oldtype)
 {
+  FAR struct tls_info_s *tls = tls_get_info();
+  int ret = OK;
+
+  /* Suppress context changes for a bit so that the flags are stable. (the
+   * flags should not change in interrupt handling).
+   */
+
+  sched_lock();
```

Comment:
I'm not sure if this is needed any longer, guarding tcb->flags makes sense, but now no one else but the thread itself writes to tl_cpstate.

Reply:
yes, I think so.